arrow_util/src/string.rs

```diff
@@ -62,6 +63,26 @@ impl<K: AsPrimitive<usize> + FromPrimitive + Zero> PackedStringArray<K> {
         id
     }
 
+    /// Extends this [`PackedStringArray`] by `range` elements from `other`
+    pub fn extend_from_range(&mut self, other: &PackedStringArray<K>, range: Range<usize>) {
+        let first_offset = other.offsets[range.start].as_();
+        let end_offset = other.offsets[range.end].as_();
```

Comment:
```suggestion
        let first_offset: usize = other.offsets[range.start].as_();
        let end_offset: usize = other.offsets[range.end].as_();
```

The `.as_()` casts are a bit hard to read w/o type annotations. In fact the types of these two variables is only constraint by the `K::from_usize` call at the very end of this function.